apps/shared/models/team.ts

```diff
@@ -0,0 +1,34 @@
+import {Team, User} from '@octokit/webhooks-types';
+import {GithubBaseModel, GithubHelperFunctions} from './base';
+
+export interface FirestoreTeam {
+  readonly slug: string;
+  readonly name: string;
+  readonly privacy: Team['privacy'];
+}
+
+export class GithubTeam extends GithubBaseModel<FirestoreTeam> {
+  readonly slug = this.data.slug;
+  readonly name = this.data.name;
+  readonly private = this.data.privacy;
+
+  static override githubHelpers: GithubHelperFunctions<Team, FirestoreTeam> = {
+    buildRefString(model: Team) {
+      return `githubTeam/${model.node_id}`;
+    },
+    fromGithub(model: Team) {
+      return {
+        name: model.name,
+        privacy: model.privacy,
+        slug: model.slug,
+      };
+    },
+  };
+}
+
+export function isTeam(userOrTeam: User | Team): userOrTeam is Team {
```

Comment:
Nit: Suggestion to make it obvious this is for the github webhook payload identification (same for `isUser`)

```suggestion
export function isTeamPayload(userOrTeam: User | Team): userOrTeam is Team {
```

Reply:
I went with `is`<Type>FromGithub`, Its not actually the payload from a user or team event, as the object appears throughout the webhook payloads when the objects are present.